workspaces/mujoco_workspace.py

```diff
@@ -77,16 +81,28 @@ def train(self):
                 self.save_snapshot()
 
             if done:
+                values = np.array(val_seq)
+                ret = np.array(rew)
+                for i in range(len(rew)):
+                    s = 0
+                    for j in range(len(rew) - i):
```

Comment:
as I have told you, you can avoid O(N^2) algorithm if you go backwards and do
`ret[i] = rew[i] + gamma*ret[i+1]`